### PR TITLE
[Mellanox] [reboot] [asan] stop asan-enabled containers on reboot

### PR DESCRIPTION
**Pull request by Yakiv-Huryk**
During a cold reboot on an image built with ENABLE_ASAN=y, swss and syncd (only on Mellanox platform) are explicitly stopped.
This will make the relevant processes receive SIGTERM and generate the ASAN reports.
The behavior for fast/warm reboots is not affected.
The behavior of regular (ENABLE_ASAN=n) images is not affected.

#### What I did
Modified the reboot script so the asan logs are generated on reboot.
#### How I did it
Added stop_services_asan() to reboot script.
#### How to verify it
Added "asan: 'yes'" to the /etc/sonic/sonic_version.yml
Checked that relevant containers are stopped and logs are generated

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@Yakiv-Huryk  could you please add to the PR description the behavior expected? what will be done in case of reboot/warm/fast? beside of that LGTM

**Review by liat-grozovik — APPROVED** [review]

File: scripts/reboot [hunk 1]
```diff
@@ -78,6 +79,17 @@ function stop_sonic_services()
     stop_pmon_service
 }
 
+function stop_services_asan()
+{
+    if [[ x"$ASIC_TYPE" == x"mellanox" ]]; then
+        debug "Stopping syncd for ASAN"
+        systemctl stop syncd
```
Comment (theasianpianist) [inline scripts/reboot:86]: Stopping SWSS should also stop syncd automatically, is it necessary to explicitly stop syncd here?

Comment (Yakiv-Huryk) [reply scripts/reboot]: Good catch, thanks. I've updated the PR: https://github.com/Azure/sonic-utilities/pull/2107/commits/5484b43d83a50de6217c63cef9290fe9830c2068

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@theasianpianist can you please signoff?

File: scripts/reboot [hunk 2]
```diff
@@ -78,6 +79,12 @@ function stop_sonic_services()
     stop_pmon_service
 }
 
+function stop_services_asan()
+{
+    debug "Stopping swss for ASAN"
+    systemctl stop swss
```
Comment (theasianpianist) [inline scripts/reboot:85]: Want to confirm here that SWSS needs to be stopped on all platforms, not just on Mellanox

Comment (Yakiv-Huryk) [reply scripts/reboot:85]: It's for all platforms since asan for swss is platform-independent.

**Comment by liat-grozovik** [comment]
@theasianpianist kindly reminder to signoff

**Review by theasianpianist — APPROVED** [review]

**Pull request merged**